public/app/features/logs/components/LogRowContext.tsx

```diff
@@ -74,31 +80,54 @@ interface LogRowContextGroupHeaderProps {
   row: LogRowModel;
   rows: Array<string | DataQueryError>;
   onLoadMoreContext: () => void;
+  groupPosition: LogGroupPosition;
   shouldScrollToBottom?: boolean;
   canLoadMoreRows?: boolean;
+  logsSortOrder?: LogsSortOrder | null;
 }
 interface LogRowContextGroupProps extends LogRowContextGroupHeaderProps {
   rows: Array<string | DataQueryError>;
+  groupPosition: LogGroupPosition;
   className?: string;
   error?: string;
+  logsSortOrder?: LogsSortOrder | null;
 }
 
 const LogRowContextGroupHeader: React.FunctionComponent<LogRowContextGroupHeaderProps> = ({
   row,
   rows,
   onLoadMoreContext,
   canLoadMoreRows,
+  groupPosition,
+  logsSortOrder,
 }) => {
   const { header } = useStyles2(getLogRowContextStyles);
 
+  // determine the position in time for this LogGroup by taking the ordering of
+  // logs and position of the component itself into account.
+  let logGroupPosition = '';
```

Comment:
Not really in love with this block.

What about:

```javascript
let logGroupPosition = 'after';
if (groupPosition === LogGroupPosition.Bottom) {
  if (logsSortOrder === LogsSortOrder.Descending) {
    logGroupPosition = 'before';
  }
} else if (logsSortOrder === LogsSortOrder.Ascending) {
  logGroupPosition = 'before';
}
```

Reply:
I was looking at those (my) `if`s for 10minutes and did not came up with that. Thanks a lot, that looks way better! 

Fixed in https://github.com/grafana/grafana/pull/55475/commits/6018cbe4ef9c7e2882f89ad89e1b420bad935a64.